Library/Homebrew/tap.rb

```diff
@@ -842,15 +842,25 @@ def ==(other)
   end
 
   def self.each(&block)
-    return unless TAP_DIRECTORY.directory?
-
     return to_enum unless block
 
-    TAP_DIRECTORY.subdirs.each do |user|
-      user.subdirs.each do |repo|
-        yield fetch(user.basename.to_s, repo.basename.to_s)
-      end
+    installed_taps = if TAP_DIRECTORY.directory?
+      TAP_DIRECTORY.subdirs
+                   .flat_map(&:subdirs)
+                   .map(&method(:from_path))
+    else
+      []
     end
+
+    available_taps = if Homebrew::EnvConfig.no_install_from_api?
+      installed_taps
+    else
+      default_taps = [CoreTap.instance]
+      default_taps << CoreCaskTap.instance if OS.mac? # rubocop:disable Homebrew/MoveToExtendOS
+      installed_taps + default_taps
```

Comment:
I'm pretty sure this will include the core taps twice each if they are installed locally and we are using the API. Admittedly that's not how most users use Homebrew anymore but that's still a possible combination.

Reply:
That's why there's a `.uniq` at the end.

Reply:
🤦 Of course, my bad.